File: orika-spring-boot-starter/src/main/java/net/rakugakibox/spring/boot/orika/OrikaProperties.java
package net.rakugakibox.spring.boot.orika;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * The configuration properties for Orika.
 */
@Data
@ConfigurationProperties("orika")
public class OrikaProperties {

    /**
     * Whether to enable auto-configuration.
     */
    private boolean enabled = true;

    /**
     * Whether to use built-in converters (MapperFactoryBuilder#useBuiltinConverters(boolean)).
     * Follows Orika's behavior by default.
     */
    private Boolean useBuiltinConverters;

    /**
     * Whether to use auto-mapping (MapperFactoryBuilder#useAutoMapping(boolean)).
     * Follows Orika's behavior by default.
     */
    private Boolean useAutoMapping;

    /**
     * Whether to map null values (MapperFactoryBuilder#mapNulls(boolean)).
     * Follows Orika's behavior by default.
     */
    private Boolean mapNulls;

    /**
     * Whether to dump the current state of the mapping infrastructure objects
     * upon occurrence of an exception while mapping (MapperFactoryBuilder#dumpStateOnException(boolean)).
     * Follows Orika's behavior by default.
     */
    private Boolean dumpStateOnException;

    /**
     * Whether the class-map should be considered 'abstract' (MapperFactoryBuilder#favorExtension(boolean)).
     * Follows Orika's behavior by default.
     */
    private Boolean favorExtension;

    /**
     * Whether full field context should be captured (MapperFactoryBuilder#captureFieldContext(boolean)).
     * Follows Orika's behavior by default.
     */
    private Boolean captureFieldContext;

}

File: src/main/kotlin/dev/akkinoc/spring/boot/orika/OrikaProperties.kt
package dev.akkinoc.spring.boot.orika

import ma.glasnost.orika.impl.DefaultMapperFactory.MapperFactoryBuilder
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

/**
 * The configuration properties for Orika.
 *
 * @property enabled
 *  Whether to enable auto-configuration.
 * @property useBuiltinConverters
 *  Whether to use built-in converters.
 *  See also [MapperFactoryBuilder.useBuiltinConverters].
 *  By default, follows Orika's behavior.
 * @property useAutoMapping
 *  Whether to use auto-mapping.
 *  See also [MapperFactoryBuilder.useAutoMapping].
 *  By default, follows Orika's behavior.
 * @property mapNulls
 *  Whether to map nulls.
 *  See also [MapperFactoryBuilder.mapNulls].
 *  By default, follows Orika's behavior.
 * @property dumpStateOnException
 *  Whether to dump the current state of the mapping infrastructure objects
 *  upon occurrence of an exception while mapping.
 *  See also [MapperFactoryBuilder.dumpStateOnException].
 *  By default, follows Orika's behavior.
 * @property favorExtension
 *  Whether to favor extension by default in registered class-maps.
 *  See also [MapperFactoryBuilder.favorExtension].
 *  By default, follows Orika's behavior.
 * @property captureFieldContext
 *  Whether full field context should be captured.
 *  See also [MapperFactoryBuilder.captureFieldContext].
 *  By default, follows Orika's behavior.
 */
@ConfigurationProperties("orika")
@ConstructorBinding
data class OrikaProperties(
        val enabled: Boolean = true,
        val useBuiltinConverters: Boolean? = null,
        val useAutoMapping: Boolean? = null,
        val mapNulls: Boolean? = null,
        val dumpStateOnException: Boolean? = null,
        val favorExtension: Boolean? = null,
        val captureFieldContext: Boolean? = null,
)
